winch/codegen/src/isa/aarch64/masm.rs

```diff
@@ -257,15 +258,9 @@ impl Masm for MacroAssembler {
         }
     }
 
-    fn checked_uadd(
-        &mut self,
-        _dst: Reg,
-        _lhs: Reg,
-        _rhs: RegImm,
-        _size: OperandSize,
-        _trap: TrapCode,
-    ) {
-        todo!()
+    fn checked_uadd(&mut self, dst: Reg, lhs: Reg, rhs: RegImm, size: OperandSize, trap: TrapCode) {
+        self.add(dst, lhs, rhs, size);
+        self.asm.trapif(Cond::Vs, trap);
```

Comment:
I believe here we'd want to use `Cond::Hs`? The `Cond::Vs` is for signed overflow. (Ref: https://developer.arm.com/documentation/dui0801/l/Condition-Codes/Carry-flag)